File: algorithms/S/Slowest Key/Slowest Key.java
class Solution {
    public char slowestKey(int[] releaseTimes, String keysPressed) {
        int max = releaseTimes[0];
        char ch = keysPressed.charAt(0);
        for(int i=1;i<releaseTimes.length;i++){
            int diff = releaseTimes[i]-releaseTimes[i-1];
            if( diff >= max){
                if(diff>max)
                    ch = keysPressed.charAt(i);
                else if(diff== max)
                    ch = (char)Math.max((int) ch, (int) keysPressed.charAt(i));
                max = diff;
            }   
        }
        return ch;      
    }    
}

File: scripts/algorithms/S/Slowest Key/Slowest Key.py
# Runtime: 111 ms (Top 25.49%) | Memory: 14.1 MB (Top 45.43%)
class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        max_dur = releaseTimes[0]
        max_key = keysPressed[0]

        for i in range(1, len(releaseTimes)):
            if releaseTimes[i] - releaseTimes[i-1] > max_dur:
                max_dur = releaseTimes[i] - releaseTimes[i-1]
                max_key = keysPressed[i]
            elif releaseTimes[i] - releaseTimes[i-1] == max_dur and max_key < keysPressed[i]:
                 max_key = keysPressed[i]

        return max_key